2026 First Contest Bronze: Cow Splits Solution (#5938)
cpinitiative/usaco-guide: solutions/bronze/usaco-1540.mdx

---
id: usaco-1540
source: 2026 First Contest Bronze
title: Cow Splits
author: Aarav Sethi
---

[Official Analysis (C++)](https://usaco.org/current/data/sol_prob2_bronze_season26contest1.html)

## Explanation

### Observation 1:

> $M=0$ if and only if the string must already be square. For example, WCOCOWWCOCOW is WCOCOW repeated twice, requiring no splits.

Reviewer: ```suggestion
$M=1$ if and only if the string must already be square. For example, `WCOCOWWCOCOW` is `WCOCOW` repeated twice, allowing for a single deletion operation.
```

### Observation 2:

If $N$ is odd, no solution exists. Square strings have even length, so odd $N*3$ cannot be formed from even-length square strings. Therefore, $N$ must be even.

### Observation 3:

> $M \le 3$ is always true, because a valid solution is always just grouping all the C’s, O’s, and W’s.

Reviewer: why are the quotes curly 😭 

also maybe add a bit more specification to the construction, tho it's prolly fine

			<td className="bg-purple-300 dark:bg-purple-800">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">C</td>
			<td className="bg-purple-300 dark:bg-purple-800">O</td>
			<td className="bg-purple-300 dark:bg-purple-800">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">O</td>
			<td className="bg-blue-300 dark:bg-blue-600">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">C</td>
			<td className="bg-purple-300 dark:bg-purple-800">O</td>
			<td className="bg-purple-300 dark:bg-purple-800">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">C</td>
			<td className="bg-purple-300 dark:bg-purple-800">O</td>
			<td className="bg-purple-300 dark:bg-purple-800">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">C</td>
			<td className="bg-blue-300 dark:bg-blue-600">O</td>
			<td className="bg-blue-300 dark:bg-blue-600">W</td>
			<td className="bg-blue-300 dark:bg-blue-600">C</td>
		</tr>
	</tbody>
</table>
</center>

So for this operation 1 would be `CCOWCCCOWC` and operation 2 would be `OWOWOWOW`. Both valid square strings.

## Implementation

**Time Complexity:** $\mathcal{O}(N)$

<LanguageSection>

<CPPSection>

```cpp
#include <bits/stdc++.h>

> #define NEWLINE (t == 0 ? "" : "\n")  // this exists to avoid trailing newlines

Author: I think it was either spaces or newlines creating issues for me, so I just did both.

> #define NEWLINE (t == 0 ? "" : "\n")  // this exists to avoid trailing newlines

Reviewer: edi code does

```cpp
cout << ans[i] << " \n"[i == 3 * n - 1];
```

that might be a cleaner alternative

> #define NEWLINE (t == 0 ? "" : "\n")  // this exists to avoid trailing newlines

Author: Turns out it did not need checks for newlines, just spaces.

---
id: usaco-1540
source: 2026 First Contest Bronze
title: Cow Splits
author: Aarav Sethi
---

[Official Analysis (C++)](https://usaco.org/current/data/sol_prob2_bronze_season26contest1.html)

## Explanation

### Observation 1:

$M=1$ if and only if the string must already be square. For example, `WCOCOWWCOCOW` is `WCOCOW` repeated twice, allowing for a single deletion operation.

### Observation 2:

If $N$ is odd, no solution exists. Square strings have even length, so odd $N*3$ cannot be formed from even-length square strings. Therefore, $N$ must be even.

### Observation 3:

$M \le 3$ is always true, because a valid solution is always just grouping all the `C`'s, `O`'s, and `W`'s.

### Observation 4:

> $M \le 2$ is always true. This is because any two possible substrings (`COW`, `OWC`, `WCO`) will all have a substring atleast 2 characters long in common.

Reviewer: ```suggestion
$M \le 2$ is always true. This is because any two possible substrings (`COW`, `OWC`, `WCO`) share a common substring of at least length 2.
```

---
id: usaco-1540
source: 2026 First Contest Bronze
title: Cow Splits
author: Aarav Sethi
---

[Official Analysis (C++, Java, Python)](https://usaco.org/current/data/sol_prob2_bronze_season26contest1.html)

## Explanation

### Observation 1:

$M=1$ if and only if the string is already square. For example, `WCOCOWWCOCOW` is `WCOCOW` repeated twice, allowing for a single deletion operation.

### Observation 2:

> If $N$ is odd, no solution exists. Square strings have even length, so odd $N*3$ cannot be formed from even-length square strings. Therefore, $N$ must be even.

Reviewer 2: If N is odd, no solution exists. Square strings have even length, and since the total length $N * 3$ will be odd, it cannot be formed from even-length square strings. Therefore, $N$ must be even.

### Observation 3:

> $M \le 3$ is always true, because a valid solution is always just grouping all the `C`'s, `O`'s, and `W`'s.

Reviewer 2: ```suggestion
$M \le 3$ is always true, because a valid solution is just grouping all the `C`'s, `O`'s, and `W`'s in separate operations.
```

### Observation 4:

$M \le 2$ is always true. This is because any two possible substrings (`COW`, `OWC`, `WCO`) share a common substring of at least length 2.

For example `COW` and `OWC` have `OW` in common, or `WCO` and `OWC` have `WC` in common.

Using this property a greedy algorithm can be made by:

> - Split the string in half

Reviewer 3: isn't it splitting

> - For both halfs of the string, read all $N/2$ components left to right.

Reviewer 2: ```suggestion
- For both halves of the string, read all $N/2$ components left to right.
```

- Add the largest common substring of both components into operation 1, and the remainder into operation 2.

> Since each operation are having the same characters added for every 3 characters from both halves of the string, each operation will end up as a square string.

Reviewer 2: ```suggestion
Since each operation has the same characters added for every corresponding component from both halves of the string, each operation will end up as a square string.
```
